packages/redux/src/raffles/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,240 @@
+import * as actionTypes from '../actionTypes';
+import * as fromReducer from './../reducer';
+import {
+  mockRaffleEstimationResponse,
+  mockRaffleParticipationResponse,
+  mockRaffleResponse,
+  mockRafflesResponse,
+  participationId_0,
+  participationId_1,
+  raffleId,
+} from 'tests/__fixtures__/raffles';
+import {
+  RaffleParticipation,
+  toBlackoutError,
+} from '@farfetch/blackout-client';
+import reducer, { entitiesMapper } from '../reducer';
+import type { RaffleEntity } from '../../entities';
+import type { RafflesState } from '../types';
+
+let initialState: RafflesState;
+
+describe('raffles reducer', () => {
+  beforeEach(() => {
+    initialState = fromReducer.INITIAL_STATE;
+  });
+  describe('loading() reducer', () => {
+    it.each([
+      actionTypes.FETCH_RAFFLES_REQUEST,
+      actionTypes.FETCH_RAFFLE_REQUEST,
+      actionTypes.FETCH_RAFFLE_ESTIMATION_REQUEST,
+      actionTypes.FETCH_RAFFLE_PARTICIPATION_REQUEST,
+    ])('should handle %s action type', actionType => {
+      const state = reducer(undefined, { type: actionType });
+      expect(state.isLoading).toBe(true);
+      expect(state.error).toBeNull();
+    });
+  });
+  describe('error() reducer', () => {
+    it.each([
+      actionTypes.FETCH_RAFFLES_FAILURE,
+      actionTypes.FETCH_RAFFLE_FAILURE,
+      actionTypes.FETCH_RAFFLE_ESTIMATION_FAILURE,
+      actionTypes.FETCH_RAFFLE_PARTICIPATION_FAILURE,
+    ])('should handle %s action type', actionType => {
+      const action = {
+        type: actionType,
+        payload: {
+          error: toBlackoutError(new Error('dummy error')),
+        },
+      };
+      const state = reducer(undefined, action);
+      expect(state.isLoading).toBe(false);
+      expect(state.error).toBe(action.payload.error);
+    });
+  });
+  describe('result() reducer', () => {
```

Comment:
This file needs a line separating each `describe` and `it` block, as they are all together and makes it hard to read.

Reply:
done